client-src/elements/chromedash-feature-row.js

```diff
@@ -184,76 +142,53 @@ class ChromedashFeatureRow extends LitElement {
-    activeOwners.sort();
-    return activeOwners;
+  isActiveGate(gate) {
+    return gate.state == 2 || gate.state == 3 || gate.state == 4;
   }
 
-  getActiveApprovals(feature) {
-    const featureApprovals = this.approvals[feature.id];
-    // TODO(jrobbins): Limit to only owners of active intents
-    const activeApprovals = featureApprovals;
-    return activeApprovals;
+  getActiveStages(feature) {
+    const featureGates = this.gates[feature.id] || [];
+    const activeGates = featureGates.filter(g => this.isActiveGate(g));
+    const activeStageIds = new Set(activeGates.map(g => g.stage_id));
+    const activeStagesAndTheirGates = [];
+    for (const activeStageId of activeStageIds) {
+      activeStagesAndTheirGates.push({
+        stageId: activeStageId,
+        gates: featureGates.filter(g => g.stage_id == activeStageId),
+      });
+    }
+    console.log(activeStagesAndTheirGates);
+    return activeStagesAndTheirGates;
   }
 
-  renderApprovalsSoFar(approvals) {
-    const result = [];
-    for (const stateItem of STATE_NAMES) {
-      const state = stateItem[0];
-      const stateName = stateItem[1];
-      const approvalsWithThatState = approvals.filter(a => a.state == state);
-      const setters = approvalsWithThatState.map(a => a.set_by.split('@')[0]);
-      if (setters.length > 0) {
-        result.push(html`<span>${stateName}: ${setters.join(', ')}. </span>`);
-      }
-    }
-    return result;
+  renderActiveStageAndGates(stageAndGates) {
+    const stageName = ''; // TODO(jrobbins) get this data.
+    return html`
+      <div>
+        ${stageName}
+        ${stageAndGates.gates.map(gate => html`
+        <chromedash-gate-chip
+          .feature=${this.feature}
+          .gate=${gate}
+        ></chromedash-gate-chip>`)}
+      </div>
+    `;
   }
 
   renderHighlights(feature) {
     if (this.columns == 'approvals') {
-      const nextReviewDate = this.getEarliestReviewDate(feature);
-      const owners = this.getActiveOwners(feature);
-      const activeApprovals = this.getActiveApprovals(feature);
-      // TODO(jrobbins): show additional_review.
+      const activeStages = this.getActiveStages(feature);
+      // TODO(jrobbins): group gates by stage
 
       return html`
         <div class="highlights">
-          ${nextReviewDate ? html`
-            <div>
-              Next review date: ${nextReviewDate}
-            </div>
-            ` : nothing}
-          ${owners.length == 1 ? html`
-            <div>
-              Owner: ${owners[0]}
-            </div>
-            ` : nothing}
-          ${owners.length > 1 ? html`
-            <div>
-              Owners: ${owners.join(', ')}
-            </div>
-            ` : nothing}
-          ${activeApprovals && activeApprovals.length > 0 ? html`
-            <div>
-              ${this.renderApprovalsSoFar(activeApprovals)}
-            </div>
-            ` : nothing}
+          ${activeStages.length > 0 ? html`
+          <div>
```

Comment:
Correct HTML indentation here?
```suggestion
            <div>
```

Reply:
Fixed.